src/dune_rules/command.mli

```diff
@@ -72,21 +72,19 @@ module Args : sig
   val as_any : without_targets t -> any t
 end
 
-(* TODO: Using list in [with_targets t list] complicates the API unnecessarily:
-   we can use the constructor [S] to concatenate lists instead. *)
 val run
   :  dir:Path.t
   -> ?sandbox:Sandbox_config.t
   -> ?stdout_to:Path.Build.t
   -> Action.Prog.t
-  -> Args.any Args.t list
+  -> Args.any Args.t
```

Comment:
I definitely prefer this as it now encourages the right way to think about arguments. It's pretty evident that everybody was confused whether to use `Args.t list` or `Args.t` before.

Reply:
But maybe I spoke to soon since it makes calling `run` almost always more awkward.